backend/README.md

```diff
@@ -1,4 +1,60 @@
+in the route itself for the ability to "enroll" them into a class.
+
+### Structure
+
+user_router := router.Group("/user"){
+
+    user_router.POST("/register", handleUserRegister) // The request respondst to the url matching "/user/register" and should include
+    // information such as first_name, last_name, email. How is user_id being generated? will it be given to us in the request or is is
+    // it made automatically in the database and sent to the front end in a different call
+
+    user_router.POST("/login", handleUserLogin) // The request responds to the url matching "/user/login" and should include
+    // parameters we require for authentication (this will be done later), we can send the user_id to the front end as a response here
+
+    Should we add a reset password for a user route when we add authentication ?
+
+}
+
+The following are all routes for a specific user based on their user_id. Certain routes should only be accessible to professors, e.g. course creation, and archive course.
+
+specific_user_router := router.Group("/user/:user_id"){
+
+    user_id is a prerequiste for all requests below
+
+    specific_user_router.POST("/enroll", handleEnrollment) // The request responds to url matching: "/user/:user_id/enroll?course=:course_id"
+
+    specific_user_router.POST("/:course_id/createPost", handleCreatePost) // The request responds to url matching "/user/:user_id/:course_id/createPost"
+
+    specific_user_router.POST("/:course_id/replyToPost", handleReplyToPost) // This route can be used for a user replying to posts made,
+    // the request should include information about the post that can be used to find it (thread_id?)
+
+    specific_user_router.POST("/:course_id/:thread_id/createComment", handleComment) // The request responds to the url matching: "/user/:user_id/:course_id/:thread_id/createComment"
+    // Since each comment is specific to a thread in a specific course we will require course_id, and thread_id for comments along with the prerequiste user_id as well
+
+    specific_user_router.DELETE("/:course_id/:thread_id", handleDeletePost) // The request responds to the url matching "/users/:user_id/:course_id/:thread_id/deletePost"
+    // Middleware will be required to make sure only the author of the post, or an instructor is able to delete the post
+    // for author authentication we can check the user_id supplied with the author entry of the thread
+
+    specific_user_router.PUT("/:course_id/:thread_id/:comment_id/upVote", handleUpVote) // The request responds to the url matching "/users/:user_id/:course_id/:thread_id/:comment_id/upVote"
+    // This is an update method that allows users to upvote comments on a given post.
+    // PUT can be replaced by PATCH, since both can be used for Updating, but PATCH seems to be for modfiying, maybe use PATCH here instead?
```

Comment:
some typos here:

Updating -> updating
modfiying -> modifying

in regards to the question: i agree that PATCH would be better than PUT based on the definition here:
```
PUT method uses the request URI to supply a modified version of the requested resource which replaces the original version of the resource, whereas the PATCH method supplies a set of instructions to modify the resource.
```